sklearn/ensemble/tests/test_forest.py

```diff
@@ -228,18 +231,20 @@ def check_importances(name, criterion, X, y):
         assert_less(np.abs(importances - importances_bis).mean(), 0.001)
```

Comment:
Apparently only the regressors with "mae" fail here. Classifiers and other regression scores pass. It's probably the case that mae is more sensitive to changes in scale of the weights. I think it's ok to increase the threshold. I did some experiments in an interactive shell and the first three features (the informative onces) are always more important that the others.

Maybe you can replace this check by the following:

```python
# The forest estimator can detect that only the first 3 features of the dataset
# are informative:
expected_mask = np.zeros(20, dtype=bool)
expected_mask[:3] = True
assert_array_equal(est.feature_importances_ > 0.1, expected_mask)
```

You call also add a check to rescale X instead of the `sample_weight` array and check that the features importances are exactly equal to the reference model.